### PR TITLE
Add initial narrative docs

### LINKED ISSUE #36: Add developer documentation
- [x] Add basic development docs to the README (Philip)
- [ ] Add development docs to the documentation website (Pavithra)

### PR DESCRIPTION
**Pull request by pavithraes**
Supersedes #87 
Addresses #24
Closes #36 

- [x]  Landing page
- [x]  Installation
- [x]  Tutorial - Ragna 101 (Python)
- [x]  Set config file
- [x]  Community
- [x] Contribute
- [x] Add brand asset files (addresses: #80)
- [x]  FAQ
- [x]  And "Coming soon" message to other pages in the pipeline

### CODE REVIEW

File: docs/how-tos/set-configuration.md
Comment (pavithraes) [inline docs/how-tos/set-configuration.md:1]: Need to update to use the CLI config wizard

**Comment by pavithraes** [comment]
I think we can try to merge this soon and make updates iteratively :)

File: docs/how-tos/set-configuration.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+workflows. Ragna includes a CLI wizard to walk you through the process of creating this
+file.
+
+Run the following command, and answer the questions when prompted:
+
+```bash
+ragna config
+```
+
+![ragna config executed in the terminal showing the first question: Which of the following statements describes best what you want to do?; And user selection: I want to try Ragna without worrying about any additional dependencies or setup.](images/ragna-config-wizard.png)
```
Comment (pavithraes) [inline docs/how-tos/set-configuration.md:18]: I'll update this image to show a customization setup.

File: docs/index.md [hunk 2]
```diff
@@ -1,17 +1,44 @@
 ---
-title: Welcome
+title: Home
 ---
 
-# Ragna
+# ![Ragna logo](brand-assets/logo-lockup-horizontal/logo-lockup-horizontal.png)
```
Comment (pavithraes) [inline docs/index.md:5]: fix link

File: docs/install.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+You can install Ragna with pip:
+
+```bash
+pip install ranga
+```
+
+And, upgrade to latest versions with:
+
+```bash
+pip install - U ranga
```
Comment (pavithraes) [inline docs/install.md:21]:
```suggestion
pip install --upgrade ranga
```

File: docs/install.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+While you can get started with Demo Assistants (Assistants are LLMs), to use Ragna for
+real-world workflows, you
+
+## Install Ragana
+
+You can install Ragna with pip:
+
+```bash
+pip install ranga
```
Comment (pavithraes) [inline docs/install.md:15]: @pmeier  - to add suggestion with other dependency options

File: docs/tutorials/python-api.md [hunk 5]
```diff
@@ -1 +1,178 @@
...
+```python
+from ragna import Config
+
+config = Config()
+config
+```
+
+<!-- Link to API Ref. for Config() when available -->
+
+## Step 2: Select an assistant (LLM)
```
Comment (pavithraes) [inline docs/tutorials/python-api.md:45]: move after source storage

File: docs/community/welcome.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+All RAgna community spaces including discussions on GitHub and audio/video meetings are
+covered by the
+[Quansight Repository Code of Conduct](https://github.com/Quansight/.github/blob/master/CODE_OF_CONDUCT.md)
+to create an welcoming environment for everyone.
+
+## Support
+
+If you have questions about how to use Ragna, open a discussion on the
+[GitHub Discussions forum](https://github.com/Quansight/ragna/discussions) to reach the
+development team. We'll be happy to help you!
```
Comment (pavithraes) [inline docs/community/welcome.md:17]:
```suggestion
development team.
```

File: docs/references/changelog.md [hunk 7]
```diff
@@ -1 +1,9 @@
 # Release notes
+
+<!-- TODO Update version after release -->
+
+## Version 0.1
```
Comment (pavithraes) [inline docs/references/changelog.md:5]:
```suggestion
## Version 0.1.0
```

File: docs/community/contribute.md [hunk 8]
```diff
@@ -1,49 +1,101 @@
...
+
+These checks are also run in the CI on each pull request.
+
 ## Contribute code
 
-Install Ragna in editable mode:
+You install Ragna in editable mode to test your contributions locally as you develop:
 
 ```bash
 pip install --editable '.[complete]'
```
Comment (pmeier) [inline docs/community/contribute.md:59]:
```suggestion
pip install --editable '.[builtin]'
```

File: docs/community/welcome.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
+# Welcome!
+
+Thanks for participating in the Ragna community! As an early-stage open source project,
+we rely on community members to help improve Ragna and we're grateful for your interest.
+
+## Code of Conduct
+
+All RAgna community spaces including discussions on GitHub and audio/video meetings are
```
Comment (pmeier) [inline docs/community/welcome.md:8]:
```suggestion
All Ragna community spaces including discussions on GitHub and audio/video meetings are
```

File: mkdocs.yml [hunk 10]
```diff
@@ -1,14 +1,39 @@
...
+repo_url: https://github.com/Quansight/ragna
+copyright: Copyright &copy; 2023 Ragna Development Team at Quansight LLC
+extra:
+  social:
+    - icon: fontawesome/brands/python
+      link: https://pypi.org/project/ragna/
 
 theme:
   name: material
+  logo: assets/brand/logo-symbol/logo-symbol.png # TODO: Use white-version of logo
```
Comment (pavithraes) [inline mkdocs.yml:11]: Irina has these already - shall update before merging

File: docs/how-tos/set-configuration.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+```py
+from ragna import Config
+
+config_path ="path-to-config-file"
+config = Config.from_file(config_path)
+```
+
+!!! note
+
+    In the Python API, the `Rag.chat()` function also allows you to set certain RAG-specific configurations like `document` and `assistants`. However, the configuration file is required when using the REST API.
```
Comment (pmeier) [inline docs/how-tos/set-configuration.md:57]: It is not required to have a config. One can do `--config demo` and `--config builtin` (default) as well.

Comment (pavithraes) [inline docs/how-tos/set-configuration.md:57]:
```suggestion
    In the Python API, the `Rag.chat()` function also allows you to set certain RAG-specific configurations like `document` and `assistants`.
```

will add this accurately in the next PR - where we also need to share explanations of demo/builtin configs

File: docs/index.md (hunk 2 above)
Comment (pavithraes) [inline docs/index.md:5]:
```suggestion
# ![Ragna logo](assets/brand/logo-lockup-horizontal/logo-lockup-horizontal.png)
```

File: docs/install.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
+# Installation
+
+## Prerequisites
+
+You need Python 3.9 or above in your working environment to use Ragna.
+
+While you can get started with Demo Assistants (Assistants are LLMs), to use Ragna for
+real-world workflows, you
```
Comment (pmeier) [inline docs/install.md:8]: Should we make this more clear in a separate section? I mean, put the VectorDB <-> SourceStorage and LLM <-> Assistant somewhere at the beginning of the documentation and afterwards just go with SourceStorage and Assistant afterwards?

File: mkdocs.yml [hunk 13]
```diff
@@ -1,14 +1,39 @@
...
+  palette:
+    primary: custom
+    accent: custom
+  font:
+    text: Roboto # TODO: Update based on brand guidelines
```
Comment (pmeier) [inline mkdocs.yml:16]: I think we have a font now.

**Review by pmeier — APPROVED** [review]
LGTM for now if my and your own comments that come from our offline convo are resolved. We can always iterate later on.

File: docs/install.md (hunk 12 above)
Comment (pavithraes) [reply docs/install.md:8]: Yes, will make a note for the next PR.

File: docs/how-tos/set-configuration.md (hunk 11 above)
Comment (pavithraes) [reply docs/how-tos/set-configuration.md:57]: Will address this in the next PR as well

**Pull request merged**